CONTRIBUTING.md

```diff
@@ -76,26 +58,117 @@ the WSL2 subsystem with the dev container.
-but executed against the host path. This requires that the docker command that would normally be executed with
-`mega-linter-runner` be adjusted for the host path:
+Because megalinter executes in a container within the dev container the mapped paths are mapped to the container path, but executed against the host path. This requires that the docker command that would normally be executed with `mega-linter-runner` be adjusted for the host path:
 
 - Specific folder (recursively):
   `docker run --rm -v /var/run/docker.sock:/var/run/docker.sock:rw -v <host path to folder>:/tmp/lint:rw`
   `oxsecurity/megalinter-dotnet:v6`
+
 - Specific file (or comma separated files):
   `docker run --rm -v /var/run/docker.sock:/var/run/docker.sock:rw -v <host path to folder>:/tmp/lint:rw`
   `-e SKIP_CLI_LINT_MODES=project -e MEGALINTER_FILES_TO_LINT=<relative path to file(s) from folder>`
   `oxsecurity/megalinter-dotnet:v6`
 
+## Module Development
+
+`BenchPress` is designed to be extensible. You can create your own modules and contribute them to the community.
+
+A valid module must have these files;
+
+```text
+Modules/BenchPress.Azure/Public
+└───Confirm-{Resource Type Name}.ps1
+
+Modules/BenchPress.Azure/Tests/Public
+└───Confirm-{Resource Type Name}.Tests.ps1
+
+examples/{Resource Type Name}
+├───{Resource Type Name}.Tests.ps1
+├───{Resource Type Name}.bicep
+└───{Resource Type Name}_Example.md
+```
+
+- Create a new module with the required files mentioned above.
+
+- Add `Confirm-{Resource Type Name}` function to the [Modules/BenchPress.Azure/BenchPress.Azure.psd1](../Modules/BenchPress.Azure/BenchPress.Azure.psd1) file in alphabetical order.
+
+- Add `{Resource Type Name}` to the [Modules/BenchPress.Azure/Classes/ResourceType.psm1](../Modules/BenchPress.Azure/Classes/ResourceType.psm1) file in alphabetical order.
+
+- Add `. $PSScriptRoot/Confirm-{Resource Type Name}.ps1` to the beginning of the [Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1](../Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1) file in alphabetical order.
+
+- Add a section for the new resource type to the end of the [Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1](../Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1) file in alphabetical order, such as;
+
+```powershell
+"{Resource Type Name}" {
+  return Confirm-{Resource Type Name} -ResourceGroupName $ResourceGroupName -Name $ResourceName
+}
+```
+
+### Module Testing
+
+- Run the following command in the root directory of the project to create a local copy of `BenchPress` in the `bin` folder;
+
+```powershell
+.\build.ps1 -Inline
+```
+
+- Replace the `BeforeAll` section of the `examples/{Resource Type Name}/{Resource Type Name}.Tests.ps1` file with the following code;
+
+> Don't forget to replace it back before submitting a pull request.
+
+```powershell
+# old
+Import-Module Az.InfrastructureTesting
+
+# new
+Import-Module ../../bin/BenchPress.Azure.psd1
```

Comment:
Would this work with just Import-Module BenchPress.Azure ? After running build.ps1 - Import

Reply:
yes, I updated the section, also added a note that has a link to build.ps1 script commands descriptions 👍